teuthology/provision/pelagos.py

```diff
@@ -79,36 +79,39 @@ def create(self):
             raise RuntimeError("Pelagos is not configured!")
         location = None
         try:
+            params=dict(os=self.os_name, node=self.name)
             response = self.do_request('node/provision',
-                                      data={'os': self.os_name,
-                                            'node': self.name},
-                                      method='POST')
+                                data=params, method='POST')
             location = response.headers.get('Location')
-            self.log.info("Waiting for deploy to finish")
-            self.log.info("Observe location: '%s'", location)
+            self.log.debug("provision task: '%s'", location)
+            # gracefully wait till provision task gets created on pelagos
             time.sleep(2)
-            with safe_while(sleep=15, tries=60) as proceed:
+            self.log.info("Waiting for deploy to finish")
+            sleep_time=15
+            with safe_while(sleep=sleep_time, tries=60) as proceed:
                 while proceed():
-                    if not self.is_task_active(location):
-                        break
+                    if self.deploy_task_active(location):
+                        self.log.info('Sleeping %s seconds' % sleep_time)
+                        continue
+                    break
         except Exception as e:
             if location:
                 self.cancel_deploy_task(location)
             else:
-                self.log.error("No task started")
+                self.log.error("Failed to start deploy tasks!")
```

Comment:
would it make sense to log the exeception here, too? like `self.log.exception("Failed to start deploy tasks!")`

Reply:
The exception is raised again next line, so... I think it should be decided by caller.

Reply:
yeah. but you are logging just half of the information. when you need to debug it, you'll be happy to know **what** exception was raised. especially because it's just catching the generic `Exception`

Reply:
It is not me, it is author's choice. I was not going to polish the code, just to improve it no my taste. Just giving you a chance to improve it even more with your own PR.